File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    int fi(int a[],int n,int k)
    {
        int lo=0, hi=n-1, mid, r=-1;
        while(lo<=hi)
        {
            mid=(lo+hi)/2;
            if(k>a[mid])
                lo=mid+1;
            else if(k<a[mid])
                hi=mid-1;
            else{
                r=mid;
                hi=mid-1;
            }
        }
        return r;
    }

    int li(int a[],int n,int k)
    {
        int lo=0, hi=n-1, mid, r=-1;
        while(lo<=hi)
        {
            mid=(lo+hi)/2;
            if(k>a[mid])
                lo=mid+1;
            else if(k<a[mid])
                hi=mid-1;
            else{
                r=mid;
                lo=mid+1;
            }
        }
        return r;
    }

    public int[] searchRange(int[] nums, int target) {
        int a[] = new int[2];
        a[0]=fi(nums,nums.length,target);
        a[1]=li(nums,nums.length,target);
        return a;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums, target):
        
        def binary_search(nums, target, find_first):
            lo, hi, result = 0, len(nums) - 1, -1
            while lo <= hi:
                mid = (lo + hi) // 2
                if nums[mid] < target:
                    lo = mid + 1
                elif nums[mid] > target:
                    hi = mid - 1
                else:
                    result = mid
                    if find_first:
                        hi = mid - 1
                    else:
                        lo = mid + 1
            return result
        
        first_occurrence = binary_search(nums, target, True)
        last_occurrence = binary_search(nums, target, False)
        
        return [first_occurrence, last_occurrence]
